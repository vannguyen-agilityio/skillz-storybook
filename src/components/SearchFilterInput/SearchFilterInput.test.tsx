import { cleanup, render, screen } from '@testing-library/react';
import * as renderer from 'react-test-renderer';

import { SearchFilterInput } from './index';

describe('SearchFilter render', () => {
  afterEach(cleanup);

  it('should render check data-testid SearchFilter component', () => {
    render(<SearchFilterInput />);

    const searchFilter = screen.getByTestId('searchFilter');
    expect(searchFilter).toBeTruthy();
  });

  describe('jest Snapshot testing', () => {
    it('matches SearchFilter Snapshot', () => {
      const searchFilter = renderer.create(<SearchFilterInput />).toJSON();
      expect(searchFilter).toMatchSnapshot();
    });
  });
});
