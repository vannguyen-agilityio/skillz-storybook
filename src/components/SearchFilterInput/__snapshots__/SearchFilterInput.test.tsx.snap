// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilter render jest Snapshot testing matches SearchFilter Snapshot 1`] = `
<div
  className="chakra-input__group css-bx0blc"
  data-group={true}
  data-testid="searchFilter"
>
  <div
    className="chakra-input__right-element css-1tqvvhk"
    size="md"
  >
    <img
      alt="Icon Search"
      className="chakra-image css-0"
      src="assets/icons/search.svg"
    />
  </div>
  <div
    className="chakra-input__left-element css-13g1oao"
    size="md"
  >
    <label>
      <div
        className="chakra-select__wrapper css-42b2qy"
      >
        <select
          className="chakra-select css-1vpl5d2"
          data-testid="select"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
        >
          <option
            value="1"
          >
            Filter All
          </option>
          <option
            value="2"
          >
            Option 2
          </option>
          <option
            value="3"
          >
            Option 3
          </option>
        </select>
        <div
          className="chakra-select__icon-wrapper css-162mkon"
        >
          <svg
            aria-hidden={true}
            className="chakra-select__icon"
            focusable={false}
            role="presentation"
            style={
              {
                "color": "currentColor",
                "height": "1em",
                "width": "1em",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </label>
  </div>
  <input
    className="chakra-input css-1ixfre5"
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Search Course / Question..."
  />
</div>
`;
